### DISCUSSION TITLE
How can I sanitize input for file creation?

### DISCUSSION THREAD

**Discussion by danieltomasz** [chhoumann/quickadd#546] [Q&A]
I have a shortcut for creating notes with templater template, when I add something lile
"1/f distribution' the note  "f distribution" in folder "1" is created, how can I sanitize  the input to create note like "1_f distribution" or "1f  distrubution" while still having non-sanitized variable for templater to add H1 and yaml title?

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

The slash is intentional - `/` in a file name means "put it in that subfolder" - so the way to handle titles like `1/f distribution` is to derive two variables up front: a sanitized one for the path and the raw one for the note's contents. QuickAdd can do the whole thing natively, H1 and YAML included, no Templater step needed.

Make a Macro with a user script step:

```js
module.exports = async (params) => {
    const raw = await params.quickAddApi.inputPrompt("Title");
    params.variables.rawTitle = raw;
    params.variables.safeTitle = raw.replace(/[\\/:*?"<>|]/g, "_");
};
```

Then add a Nested Choice step that runs a Template choice with File Name Format `{{VALUE:safeTitle}}`, using a template file like:

```
---
title: "{{VALUE:rawTitle}}"
---
# {{VALUE:rawTitle}}
```

Variables set in a macro step are available to every later step, so `1/f distribution` becomes the file `1_f distribution.md` while the H1 and the `title` property keep the original text.

Docs: https://quickadd.obsidian.guide/docs/Choices/MacroChoice
